pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -72,12 +73,39 @@ func (ds *dockerService) RunPodSandbox(config *runtimeApi.PodSandboxConfig) (str
 	// Assume kubelet's garbage collector would remove the sandbox later, if
 	// startContainer failed.
 	err = ds.StartContainer(createResp.ID)
+	if err != nil {
+		return createResp.ID, fmt.Errorf("failed to start sandbox container for pod %q: %v", config.Metadata.GetName(), err)
+	}
+
+	// Step 4: Trigger network plugin to setup pod network
+	// skip if pod is using hostnetwork
+	if !config.GetLinux().GetNamespaceOptions().GetHostNetwork() {
```

Comment:
should check nil pointers, e.g. `config.GetLinux()!= nil && config.GetLinux(). GetNamespaceOptions() != nil `